geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/GetMemberInformationFunction.java

```diff
@@ -142,12 +139,22 @@ public void execute(FunctionContext functionContext) {
           numConnections = numConnections + status.getNumberOfConnections();
         }
         memberInfo.setClientCount(numConnections);
+
+        ServerLauncher.ServerState state = ServerLauncher.getServerState();
+        if (state != null) {
+          memberInfo.setStatus(state.getStatus().getDescription());
+        }
       } else {
         memberInfo.setServer(false);
         InternalLocator locator = InternalLocator.getLocator();
         if (locator != null) {
           memberInfo.setLocatorPort(locator.getPort());
         }
+
+        LocatorLauncher.LocatorState state = LocatorLauncher.getLocatorState();
```

Comment:
This block looks like duplicate code.

Can we create a empty final variable called `state` which is assigned within the if/else. Then outside the if/else we can `memberInfo.setStatus(state.getStatus().getDescription());`

Reply:
@jdeppe-pivotal